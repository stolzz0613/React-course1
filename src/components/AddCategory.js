import React, { useState } from 'react'
import propTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleSubmit = (e) => {
        e.preventDefault();

        inputValue.trim().length && setCategories( categories => [inputValue, ...categories]);
        setInputValue('')
    }

    return (
        <form onSubmit= { handleSubmit }>
            <input
                type='text'
                value={inputValue}
                onChange={ (e) => setInputValue( e.target.value ) }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: propTypes.func.isRequired
}
