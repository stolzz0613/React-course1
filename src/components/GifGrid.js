import React from 'react'
import { useState, useEffect } from 'react/cjs/react.development';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    useEffect(() => {
        getGifs();
    }, [])

    const getGifs = async() => {

        const url = 'https://api.giphy.com/v1/gifs/search?api_key=SYov9Cn85jWspJqa7zaPWlMWWzduX7mR&q=Rick+And+Morty&limit=10';
        const res = await fetch( url );
        const { data } = await res.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });

        setImages(gifs);
    }

    return (
        <>
            <h3>{ category }</h3>
            <div className='card-grid'>
                {
                    images.map( img => (
                        <GifGridItem
                            {...img}
                            key= { img.id }
                        />
                    ))
                }
            </div>
        </>
    )
}
