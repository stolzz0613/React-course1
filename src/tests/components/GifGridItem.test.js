import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem";

describe('Prueba en <GifGridItem/>', () => {
    test('debe mostrar el componente correctamente', () => {
        const wrapper = shallow( <GifGridItem /> );
        expect( wrapper ).toMatchSnapshot();
    })
})
