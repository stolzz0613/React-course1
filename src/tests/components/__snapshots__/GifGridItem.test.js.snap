// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGridItem/> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img />
  <p />
</div>
`;
